baseapp/baseapp.go

```diff
@@ -427,7 +436,18 @@ func (app *BaseApp) BeginBlock(req abci.RequestBeginBlock) (res abci.ResponseBeg
 	} else {
 		// In the first block, app.deliverState.ctx will already be initialized
 		// by InitChain. Context is now updated with Header information.
-		app.deliverState.ctx = app.deliverState.ctx.WithBlockHeader(req.Header).WithBlockHeight(req.Header.Height)
+		app.deliverState.ctx = app.deliverState.ctx.
+			WithBlockHeader(req.Header).
+			WithBlockHeight(req.Header.Height)
+	}
+
+	// add block gas meter
+	if app.maximumBlockGas > 0 {
+		app.deliverState.ctx = app.deliverState.ctx.
```

Comment:
Can we declare a `gasMeter` variable instead here - less duplication?

Reply:
done